src/hotspot/os/linux/stathist_linux.cpp

```diff
@@ -0,0 +1,415 @@
+    }
+    return NULL;
+  }
+
+  value_t parsed_prefixed_value(const char* prefix, size_t scale = 1) const {
+    value_t value = INVALID_VALUE;
+    const char* const s = get_prefixed_line(prefix);
+    if (s != NULL) {
+      errno = 0;
+      const char* p = s + ::strlen(prefix);
+      value = (value_t)::strtoll(p, NULL, 10);
+      if (value == 0 && errno != 0) {
+        value = INVALID_VALUE;
+      } else {
+        value *= scale;
+      }
+    }
+    return value;
+  }
+
+};
+
+struct cpu_values_t {
+  value_t user;
+  value_t nice;
+  value_t system;
+  value_t idle;
+  value_t iowait;
+  value_t steal;
+  value_t guest;
+  value_t guest_nice;
+};
+
+void parse_proc_stat_cpu_line(const char* line, cpu_values_t* out) {
+  // Note: existence of some of these values depends on kernel version
+  out->user = out->nice = out->system = out->idle = out->iowait = out->steal = out->guest = out->guest_nice =
+      INVALID_VALUE;
+  int user, nice, system, idle, iowait, irq, softirq, steal, guest, guest_nice;
+  int num = ::sscanf(line,
+      "cpu %d %d %d %d %d %d %d %d %d %d",
+      &user, &nice, &system, &idle, &iowait, &irq, &softirq, &steal, &guest, &guest_nice);
+  if (num >= 4) {
+    out->user = user;
+    out->nice = nice;
+    out->system = system;
+    out->idle = idle;
+    if (num >= 5) { // iowait (5) (since Linux 2.5.41)
+      out->iowait = iowait;
+      if (num >= 8) { // steal (8) (since Linux 2.6.11)
+        out->steal = steal;
+        if (num >= 9) { // guest (9) (since Linux 2.6.24)
+          out->guest = guest;
+          if (num >= 10) { // guest (9) (since Linux 2.6.33)
+            out->guest_nice = guest_nice;
+          }
+        }
+      }
+    }
+  }
+}
+
+
+/////// Columns ////////
+
+// A special class to display cpu time
+class CPUTimeColumn: public Column {
+
+  long _clk_tck;
+  int _num_cores;
+
+  int do_print(outputStream* st, value_t value, value_t last_value,
+      int last_value_age, const print_info_t* pi) const {
+    // CPU values may overflow, so the delta may be negative.
+    if (last_value > value) {
+      return 0;
+    }
+    int l = 0;
+    if (value != INVALID_VALUE && last_value != INVALID_VALUE) {
+
+      // If the last sample is less than one second old, we omit calculating the cpu
+      // usage.
+      if (last_value_age > 0) {
+
+        // Values are in ticks. Convert to ms.
+        const uint64_t value_ms = (value * 1000) / _clk_tck;
+        const uint64_t last_value_ms = (last_value * 1000) / _clk_tck;
+        const uint64_t delta_ms = value_ms - last_value_ms;
+
+        // Calculate the number of wallclock milliseconds for the delta interval...
+        const int age_ms = last_value_age * 1000;
+
+        // times number of available cores.
+        const int total_cpu_time_ms = age_ms * _num_cores;
+
+        // Put the spent cpu time in reference to the total available cpu time.
+        const float percentage = (100.0f * delta_ms) / total_cpu_time_ms;
```

Comment:
Why float and not double? It will be converted to double in the jio_sprintf call anyways.